### ISSUE TITLE
ABAP Cleaner Crashes (Endless Loop?)

### ISSUE THREAD

**Issue by fmabap** [SAP/abap-cleaner#228]
Hi,

the ABAP Cleaner crashes at this sample select, if we add the 3 pragmas.

It looks like an endless loop and we have to restart eclipse.

```abap
DATA: BEGIN OF ls_matnr,
        matnr TYPE matnr,
      END OF ls_matnr.

DATA lt_matnr LIKE STANDARD TABLE OF ls_matnr.
DATA lt_mara TYPE STANDARD TABLE OF mara.

ls_matnr-matnr = 'TEST 1234'.
INSERT ls_matnr INTO TABLE lt_matnr.

SELECT mara~*
  FROM @lt_matnr AS lt_matnr
  INNER JOIN mara
  ON mara~matnr = lt_matnr~matnr
  INTO TABLE @lt_mara
  ##db_feature_mode[itabs_in_from_clause] ##itab_db_select ##ITAB_KEY_IN_SELECT.
```
If we remove the pragmas, then it works fine.

We use the latest version 1.13.0.

It crashes also if the select rules are deactivted.

<img width="251" alt="image" src="https://github.com/SAP/abap-cleaner/assets/101019006/b264cedb-6f01-4e14-82cf-1d1ae941cc4d">

Best regards
Frank

**Comment by jmgrassau** [comment]
Hi Frank, 

thanks for opening this! Hm, unfortunately I can't reproduce this error on my system. Maybe it is related to some specific profile setting (not necessarily in the "Align SELECT …" rules). Could you attach the profile you were using (Source Code / Clean Up With Interactive ABAP Cleaner…" -> Help / Open Profiles Folder -> <profile name>.cfj)?

The issue might also be related to surrounding code (e.g. the corresponding method definition). To narrow that down, you could try to create a "minimal code example" that causes the issue. This is probably easiest with the [standalone version](https://github.com/SAP/abap-cleaner/releases), because then you wouldn't have to restart all of ADT each time. What I sometimes do is copy the whole code document to notepad, then do "trial and error" to find out which parts can be deleted and simplified to still trigger the error. 

Kind regards,
Jörg-Michael

**Comment by fmabap** [comment]
Hi Jörg-Michael,

I tried the latest standalone version and I have the same issue there with the example coding from above.

It looks like that the reason is the "Move pragmas to correct position" rule. It works fine when I deactivate it.

Please find attached the profile. It contains only this active rule.

[Profile.zip](https://github.com/SAP/abap-cleaner/files/13861313/Profile.zip)

Best regards
Frank

**Comment by jmgrassau** [comment]
Hi Frank, 

ah, excellent, now I could reproduce this! It was the combination of 
* the activated option "Move (correctly positioned) pragmas from line start to line end" with 
* multiple pragmas at line start, 
* which were at the same time already at the correct line end position! 

This bug will be fixed with the next release.

Kind regards,
Jörg-Michael

**Comment by jmgrassau** [comment]
Hi Frank,

thanks again for this bug report and sorry this took a while – this issue should now be fixed with version 1.13.1, which was just released!

Kind regards,
Jörg-Michael
